### PR TITLE
Feature/update version links

### PR DESCRIPTION
**Pull request by redhug1**
### What

Use PutVersion to update links.
Code clean up.
Copy in some of latest changes from CSV exporter, like GetS3ContentLength and structure of some functions.

### How to review

Visual inspection.

### Who can review

Anyone.

### CODE REVIEW

File: handler/handler.go [hunk 1]
```diff
@@ -543,32 +555,78 @@ func (h *CsvComplete) UploadXLSXFile(ctx context.Context, instanceID string, fil
 	return s3Location, nil
 }
 
+// GetS3ContentLength obtains an S3 file size (in number of bytes) by calling Head Object
+func (h *CsvComplete) GetS3ContentLength(ctx context.Context, e *event.CantabularCsvCreated, isPublished bool) (int, error) {
+	filename := generateS3FilenameXLSX(e)
+	if isPublished {
+		headOutput, err := h.s3Public.Head(filename)
```
Comment (DavidSubiros) [inline handler/handler.go:562]: I will need to fix the IAM permissions issue (with terraform) that I mentioned before in order to give the app users permission to do a HeadObject call. Once I do, I wil lapply it to dp-cantabular-xlsx-exporter too. Until then, you will see a 403 return code from this call in develop (it works locally with dp-compose)

File: handler/handler_test.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ const (
 	testVaultPath          = "vault-root"
 	testInstanceID         = "test-instance-id"
 	testS3Location         = "s3://myBucket/my-file.csv"
-	testDownloadServiceURL = "http://test-download-service:8200"
+	testDownloadServiceURL = "http://test-download-service:8200" //!!! is this the vault port number ?, and if so why is it in this line
```
Comment (DavidSubiros) [inline handler/handler_test.go:33]: I think this is only for testing (to validate the links are created correctly) - it can have any value.

**Review by DavidSubiros — DISMISSED** [review]
LGTM - just added a couple of comments regarding HeadObject permissions (I still need to fix terraform for it to work in develop)

File: handler/handler.go (hunk 1 above)
Comment (redhug1) [reply handler/handler.go:562]: OK.
Thanks.
I'm only getting things working in compose for the time being.

File: handler/handler_test.go (hunk 2 above)
Comment (redhug1) [reply handler/handler_test.go]: OK thanks, i'll change it to a different value, to save others asking same question.

**Review by DavidSubiros — APPROVED** [review]

**Pull request merged**